exporter/sapmexporter/exporter.go

```diff
@@ -25,53 +25,115 @@ import (
+	if err != nil {
+		return nil, err
 	}
+
 	return exporterhelper.NewTraceExporter(
 		cfg,
 		se.pushTraceData,
 		exporterhelper.WithShutdown(se.Shutdown))
 }
 
-// pushTraceData exports traces in SAPM proto and returns number of dropped spans and error if export failed
-func (se *sapmExporter) pushTraceData(ctx context.Context, td pdata.Traces) (droppedSpansCount int, err error) {
-	batches, err := jaeger.InternalTracesToJaegerProto(td)
-	if err != nil {
-		return td.SpanCount(), consumererror.Permanent(err)
+// tracesByAccessToken takes a pdata.Traces struct and will iterate through its ResourceSpans' attributes,
+// regrouping by any SFx access token label value if Config.AccessTokenPassthrough is enabled.  It will drop any
+// set token label in any case to prevent serialization.
+// It returns a map of newly constructed pdata.Traces keyed by access token, defaulting to empty string.
+func (se *sapmExporter) tracesByAccessToken(td pdata.Traces) map[string]pdata.Traces {
+	tracesByToken := make(map[string]pdata.Traces, 1)
+	resourceSpans := td.ResourceSpans()
+	for i := 0; i < resourceSpans.Len(); i++ {
+		resourceSpan := resourceSpans.At(i)
+		if resourceSpan.IsNil() {
+			// Invalid trace so nothing to export
+			continue
+		}
+
+		accessToken := ""
+		if !resourceSpan.Resource().IsNil() {
+			attrs := resourceSpan.Resource().Attributes()
+			attributeValue, ok := attrs.Get(splunk.SFxAccessTokenLabel)
+			if ok {
+				attrs.Delete(splunk.SFxAccessTokenLabel)
+				if se.config.AccessTokenPassthrough {
+					accessToken = attributeValue.StringVal()
+				}
+			}
+		}
+
+		traceForToken, ok := tracesByToken[accessToken]
+		if !ok {
+			traceForToken = pdata.NewTraces()
+			tracesByToken[accessToken] = traceForToken
+		}
+
+		// Append ResourceSpan to trace for this access token
+		traceForTokenSize := traceForToken.ResourceSpans().Len()
+		traceForToken.ResourceSpans().Resize(traceForTokenSize + 1)
+		traceForToken.ResourceSpans().At(traceForTokenSize).InitEmpty()
+		resourceSpan.CopyTo(traceForToken.ResourceSpans().At(traceForTokenSize))
 	}
-	err = se.client.Export(ctx, batches)
-	if err != nil {
-		if sendErr, ok := err.(*sapmclient.ErrSend); ok {
-			if sendErr.Permanent {
-				return 0, consumererror.Permanent(sendErr)
+
+	return tracesByToken
+}
+
+// pushTraceData exports traces in SAPM proto by associated SFx access token and returns number of dropped spans
+// and the last experienced error if any translation or export failed
+func (se *sapmExporter) pushTraceData(ctx context.Context, td pdata.Traces) (droppedSpansCount int, err error) {
+	traces := se.tracesByAccessToken(td)
+	droppedSpansCount = 0
+	for accessToken, trace := range traces {
+		batches, translateErr := jaeger.InternalTracesToJaegerProto(trace)
+		if translateErr != nil {
+			droppedSpansCount += trace.SpanCount()
+			err = consumererror.Permanent(translateErr)
+			continue
+		}
+
+		exportErr := se.client.ExportWithAccessToken(ctx, batches, accessToken)
```

Comment:
If `accessToken` is "" is the default access token used?

Reply:
Right, this occurs in the updated client's worker: https://github.com/signalfx/sapm-proto/blob/6b28b8974508e838a308eb9fbf1e624b84c67132/client/worker.go#L116